slatedb/src/compactor_executor.rs

```diff
@@ -160,8 +159,17 @@ impl TokioCompactionExecutorInner {
                     .await?;
             sr_iters.push_back(iter);
         }
-        let sr_merge_iter = MergeIterator::new(sr_iters).await?;
-        MergeIterator::new([l0_merge_iter, sr_merge_iter]).await
+        let sr_merge_iter = MergeIterator::new(sr_iters).await?.with_dedup(false);
+
+        let merge_iter = MergeIterator::new([l0_merge_iter, sr_merge_iter]).await?;
```

Comment:
Does this need a `with_dedup(false)` as well? I think it might de-dup if both l0 and sr heads have the same key (but different seqnum).

Reply:
oh you are right, this is a bug >_< fixed in bad786b